### DISCUSSION TITLE
Smoothly control contentOffset of Scroll component (ScrollView, FlatList ...) on UI thread.

### DISCUSSION THREAD

**Discussion by bolan9999** [software-mansion/react-native-reanimated#4517] [Ideas]
As you know , the big problem on React Native is : can not control the contentOffset of the Scroll component smoothly. We can only try to use `scrollTo` method . But the `scrollTo` method is not smoothly at all. In my opinion, it will be amazing if I can do it like below to control contentOffset.

```
  const ref = useAnimatedRef<Reanimated.ScrollView>();
  const offset = useScrollViewOffset(ref);
  return <Reanimted.ScrollView ref={ref}>...</<Reanimted.ScrollView>>
```

```
offset.value=withTiming(...);
```

If so ,we can do this effect on React Native. 

![header](https://github.com/software-mansion/react-native-reanimated/assets/14797923/c1570cf4-c5df-459b-a2ef-109712ce9786)

Someone tried to use `scrollTo` instead, but it is not smoothly on Android. [https://github.com/zyslife/react-native-head-tab-view](url)

**Comment by tomekzaw** [comment]
Hey @bolan9999, we're actually working on this feature, see #4416.

**Reply by bolan9999** [reply]
Great
